tgui/packages/tgui/interfaces/MaterialStand.tsx

```diff
@@ -0,0 +1,120 @@
+import { createSearch, toTitleCase } from 'common/string';
+import { useBackend, useLocalState } from '../backend';
+import { Box, Button, Input, Stack, Flex, Section } from '../components';
+import { Window } from '../layouts';
+
+type Ores = {
+  id: string;
+  name: string;
+  amount: number;
+};
+
+type Ore_images = {
+  name: string;
+  icon: string;
+};
+
+type Data = {
+  ores: Ores[];
+  ore_images: Ore_images[];
+};
+
+export const MaterialStand = (props, context) => {
+  const { act, data } = useBackend<Data>(context);
+  const { ores } = data;
+  const [searchItem, setSearchItem] = useLocalState(context, 'searchItem', '');
+  const search = createSearch(searchItem, (ores) => ores.name);
+  const ores_filtered =
+    searchItem.length > 0 ? ores.filter((ore) => search(ore)) : ores;
+  return (
+    <Window title="MaterialStand" width={550} height={400}>
+      <Window.Content>
+        <Stack fill vertical>
+          <Section>
+            <Stack.Item>
+              <Input
+                autofocus
+                position="relative"
+                mt={0.5}
+                bottom="5%"
+                height="20px"
+                width="150px"
+                placeholder="Search Ore..."
+                value={searchItem}
+                onInput={(e, value) => {
+                  setSearchItem(value);
+                }}
+                fluid
+              />
+            </Stack.Item>
+          </Section>
+          <Section title="Stock" fill scrollable>
+            <Stack.Item>
+              <Flex wrap>
+                {ores_filtered.map((ore) => (
+                  <Flex.Item key={ore.id}>
+                    <Flex direction="column" m={0.5} textAlign="center">
+                      <Flex.Item>
+                        <RetrieveIcon ore={ore} />
+                      </Flex.Item>
+                      <Flex.Item>
+                        <Orename ore_name={toTitleCase(ore.name)} />
+                      </Flex.Item>
+                      <Flex.Item>Amount: {toTitleCase(ore.amount)}</Flex.Item>
+                      <Flex.Item>
+                        <Button
+                          content="Withdraw"
+                          color="transparent"
+                          onClick={() =>
+                            act('withdraw', {
+                              reference: ore.id,
+                            })
+                          }
+                        />
+                      </Flex.Item>
+                    </Flex>
+                  </Flex.Item>
+                ))}
+              </Flex>
+            </Stack.Item>
+          </Section>
+        </Stack>
+      </Window.Content>
+    </Window>
+  );
+};
+
+const RetrieveIcon = (props, context) => {
+  const { data } = useBackend<Data>(context);
+  const { ore_images } = data;
```

Comment:
```suggestion
  const { ore_images = [] } = data;
```
I don't trust byond to send this stuff